src/relay/op/nn/sparse.cc

```diff
@@ -196,5 +196,43 @@ RELAY_REGISTER_OP("nn.sparse_transpose")
     .set_support_level(1)
     .add_type_rel("SparseTranspose", SparseTransposeRel);
 
+// relay.nn.sparse_add
+bool SparseAddRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
```

Comment:
I think you also need to check if the sparse_data dtype is the same as the dense dtype.